lib/mixpanel-ruby/consumer.rb

```diff
@@ -88,7 +94,11 @@ def send(type, message)
       form_data = {"data" => data, "verbose" => 1}
       form_data.merge!("api_key" => api_key) if api_key
 
-      response_code, response_body = request(endpoint, form_data)
+      begin
+        response_code, response_body = request(endpoint, form_data)
+      rescue Exception => e
+        raise ConnectionError.new("Could not connect to Mixpanel, with error \"#{e.message}\".")
```

Comment:
Ultimately, it's your call, but there is a lot of cool things you might be able to do with an original exception that you can't do with a stringified backtrace- particularly if that exception was exotic in some way. Since you own ConnectionError, is there a reason to not add a root_error member (again, it's possible that this is simply handled with ConnectionError#cause on construction)

@tdumitrescu comment seems pretty apt as well.